### PR TITLE
Ruby/Python/JS/Swift: Add category of Private information to shared sensitive data heuristics

### PR DESCRIPTION
**Pull request by joefarebrother**
Adds a category of private information to the shared sensitive data heuristics file.
This may result in new results for the following queries: 

- `rb/sensitive-get-query`
- `py/clear-text-storage-sensitive-data`
- `py/clear-text-logging-sensitive-data`
- `py/weak-sensitive-data-hashing`
- `js/clear-text-stotage-sensitive-data`
- `js/clear-text-logging`

### CODE REVIEW

File: swift/ql/lib/codeql/swift/security/SensitiveExprs.qll
Comment (geoffw0) [inline swift/ql/lib/codeql/swift/security/SensitiveExprs.qll:52]: I see you've excluded the new results from Swift for now.  Since they're a near duplicate of Swift's existing `class SensitivePrivateInfo` I think we should aim (now or later) to define that in terms of the new result classification instead, similarly to how `class SensitivePassword` already works.  This may add a few new results (for SSN and CCN) in the four or five Swift sensitive data queries.

File: python/ql/test/query-tests/Security/CWE-312-CleartextLogging/test.py
Comment (geoffw0) [inline python/ql/test/query-tests/Security/CWE-312-CleartextLogging/test.py:72]: These `NOT FOUND` results are good observations.  We can think about mechanisms for limiting the exclusions perhaps and finding more results in cases like these.

**Review by geoffw0 — COMMENTED** [review]
As the author of the private information regexps for Swift (`class SensitivePrivateInfo`), I'm happy to see these expressions being used more widely.  They've already had a reasonable amount of testing and iteration in Swift, and I expect they will work pretty well for other languages with minimal or no additional tuning.

File: swift/ql/lib/codeql/swift/security/internal/SensitiveDataHeuristics.qll
Comment (joefarebrother) [reply swift/ql/lib/codeql/swift/security/internal/SensitiveDataHeuristics.qll:94]: Have some MRVA runs [here](https://github.com/github/codeql-team/issues/3083#issuecomment-2102753133) for queries restricted to the new sources. Seems to mostly be TPs. 

SSN is intended to match a result in a [Ruby synthetic application](https://github.com/github/codeql-team/issues/2367).

Comment (geoffw0) [reply swift/ql/lib/codeql/swift/security/internal/SensitiveDataHeuristics.qll:94]: Yep, I'm happy that "SSN" and "CCN" are useful terms to look for, the question I'm interested in is whether the FP rate on these particular patterns is acceptable.  I don't see an easy way to find just these results in your MRVA runs, so I've started my own to answer this question in particular...

Comment (geoffw0) [reply swift/ql/lib/codeql/swift/security/internal/SensitiveDataHeuristics.qll:94]: The MRVA run completed, results for those particular terms are sparse but LGTM.  That alleviates my concern about noisy FPs!  :+1:

File: swift/ql/lib/codeql/swift/security/internal/SensitiveDataHeuristics.qll
Comment (geoffw0) [reply swift/ql/lib/codeql/swift/security/internal/SensitiveDataHeuristics.qll:104]: Resolved, see other conversation.

File: swift/ql/lib/codeql/swift/security/SensitiveExprs.qll
Comment (geoffw0) [reply swift/ql/lib/codeql/swift/security/SensitiveExprs.qll:52]: ... I don't mind doing this as follow-up actually.  I already made half the changes I'll need to run a test of the SSN and CCN terms!

File: python/ql/test/query-tests/Security/CWE-312-CleartextLogging/test.py
Comment (joefarebrother) [reply python/ql/test/query-tests/Security/CWE-312-CleartextLogging/test.py:72]: Should that be considered for this PR or as potential follow-up?

**Comment by joefarebrother** [comment]
@geoffw0 What are the next steps for this? Should I just be waiting for an approval from each language team?

**Comment by erik-krogh** [comment]
@joefarebrother: Could you summarise what changed results your evaluations showed.  
What new results appeared for which queries, do you think they look good, and there any results that look bad, and is there any noticeable performance impact.

**Comment by joefarebrother** [comment]
@erik-krogh There are new results for `rb/sensitive-get-query`, `py/clear-text-storage-sensitive-data`, and `py/clear-text-logging-sensitive-data`. and `py/weak-sensitive-data-hashing`. 

New results in ruby sensitive get looks like a few TPs from new heuristics. (The MRVA run that restricts to only new heuristics shows some apparent FPs, but they are also present in the run of the full version of the query prior to changes; and appear to be the result of some nodes being spuriously detected as calls to several different methods including those matching sensitive heuristics)

Python cleartext storage and weak sensitive hashing have a few new results for which the sources look like TPs from new heuristics.
Python cleartext logging has many new results (as the query has a lot of sinks); going through them they look like mostly TPs. I did notice just one FP from an attribute named `SSN` that was not a social security number. 
There is no noticable performance impact.

**Review by erik-krogh — APPROVED** [review]
JS 👍 

But we should probably do a followup where this implementation is moved to the shared folder (together with `CryptoAlgorithmNames`?).

File: python/ql/test/query-tests/Security/CWE-312-CleartextLogging/test.py
Comment (geoffw0) [reply python/ql/test/query-tests/Security/CWE-312-CleartextLogging/test.py:72]: As potential follow-up I'd say.  There will *always* be more improvements we can make to heuristics like this.

**Review by geoffw0 — APPROVED** [review]
> @geoffw0 What are the next steps for this? Should I just be waiting for an approval from each language team?

Yes I think so.

**Comment by alexrford** [comment]
Ruby 👍 - the new results seem pretty reasonable.

**Review by alexrford — APPROVED** [review]

**Review by RasmusWL — APPROVED** [review]
LGTM :+1:

I see this as a clear improvement of what we already have, but in the context of `py/weak-sensitive-data-hashing`, I'll note that credit card numbers have such a limited input space that they **need** computationally expensive hash functions (like passwords do) -- so that's a future improvement we should make to that query (would probably require credit card information getting its' own category instead of being part of the new private category)

File: swift/ql/lib/codeql/swift/security/SensitiveExprs.qll
Comment (geoffw0) [reply swift/ql/lib/codeql/swift/security/SensitiveExprs.qll:52]: PR for Swift: https://github.com/github/codeql/pull/16570

**Pull request merged**
